### PR TITLE
AWS Shield Subscription - finalize PR#272

### PR DESCRIPTION
**Pull request by tbugfinder**
### Description

This resource was requested from here #145 (started in #272 

Also pending discussion around integration tests as this resource is for the advanced shield subscription.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [x] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [ ] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_shield_subscription.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+title: About the aws_shield_subscription Resource
+platform: aws
+---
+
+# aws\_shield\_subscription
+
+Use the `aws_shield_subscription` InSpec audit resource to test properties of a shield subscription. 
+This resource is to be used for testing the AWS Shield Advanced subscription for an account.
```
Comment (IanMadd) [inline docs/resources/aws_shield_subscription.md:9]:
```suggestion
Use the `aws_shield_subscription` InSpec audit resource to test properties of an AWS Shield Advanced subscription.
```

File: docs/resources/aws_shield_subscription.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Syntax
+
+ An `aws_shield_subscription` resource block uses the parameter to return a shield subscription.
```
Comment (IanMadd) [inline docs/resources/aws_shield_subscription.md:13]:
```suggestion
An `aws_shield_subscription` resource block returns a Shield Advanced subscription.
```

File: docs/resources/aws_shield_subscription.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    describe aws_shield_subscription do
+      it { should exist }
+    end
+
+
+#### Parameters
+
+This resource does not expect any parameters.
+
+See also the [AWS documentation on Shield Subscriptions](https://docs.aws.amazon.com/waf/latest/developerguide/shield-chapter.html).
```
Comment (IanMadd) [inline docs/resources/aws_shield_subscription.md:24]:
```suggestion
Also, see the [AWS Shield Subscriptions documentation](https://docs.aws.amazon.com/waf/latest/developerguide/shield-chapter.html).
```

File: docs/resources/aws_shield_subscription.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+
+## Properties
+
+|Property                      | Description|
+| ---                          | --- |
+|auto\_renew                   | If ENABLED, the subscription will be automatically renewed at the end of the existing subscription period. AutoRenew can be either ENABLED or DISABLED. |
```
Comment (IanMadd) [inline docs/resources/aws_shield_subscription.md:31]:
```suggestion
|auto\_renew                   | If `ENABLED`, the subscription will be automatically renewed at the end of the existing subscription period. Valid values: `ENABLED` or `DISABLED`. |
```

File: docs/resources/aws_shield_subscription.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+|end\_time                     | The date and time your subscription will end. |
+|limits                        | Specifies how many protections of a given type you can create. This is an array containing the Type of protection and the maximum number of protections that can be created for the specified Type. |
+|proactive\_engagement\_status | If ENABLED, the DDoS Response Team (DRT) will use email and phone to notify contacts about escalations to the DRT and to initiate proactive customer support. <br/> If PENDING, you have requested proactive engagement and the request is pending. The status changes to ENABLED when your request is fully processed. <br/> If DISABLED, the DRT will not proactively notify contacts about escalations or to initiate proactive customer support. |
```
Comment (IanMadd) [inline docs/resources/aws_shield_subscription.md:34]:
```suggestion
|proactive\_engagement\_status | Valid values: `ENABLED`, `DISABLED`, `PENDING`. <br> If ENABLED, the DDoS Response Team (DRT) will use email and phone to notify contacts about escalations to the DRT and to initiate proactive customer support. <br/> If `PENDING`, you have requested proactive engagement and the request is pending. The status changes to `ENABLED` when your request is fully processed. <br/> If `DISABLED`, the DRT will not proactively notify contacts about escalations or to initiate proactive customer support. |
```

File: docs/resources/aws_shield_subscription.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+|start\_time                   | The start time of the subscription, in Unix time in seconds. |
+|time\_commitment\_in\_seconds | The length, in seconds, of the AWS Shield Advanced subscription for the account. |
+
+For a comprehensive list of properties available, see [the API reference documentation](https://docs.aws.amazon.com/waf/latest/DDOSAPIReference/API_Subscription.html)
+
+## Examples
+
+##### Check Auto Renew of a Shield Subscription
```
Comment (IanMadd) [inline docs/resources/aws_shield_subscription.md:42]:
```suggestion
### Check the automatic renewal status of a Shield Subscription
```

File: docs/resources/aws_shield_subscription.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      its('auto_renew')  { should eq 'ENABLED' }
+    end
+
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers, please visit our [matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+## AWS Permissions
+
+Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `shield:DescribeSubscription` action with Effect set to Allow.
```
Comment (IanMadd) [inline docs/resources/aws_shield_subscription.md:54]:
```suggestion
Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `shield:DescribeSubscription` action with `Effect` set to `Allow`.
```

**Review by IanMadd — COMMENTED** [review]
Docs editing

Also, could you add a link to the new page in the README.md

**Comment by sathish-progress** [comment]
@tbugfinder  thanks for the PR. Could you please add valid integration tests for this ..

**Comment by tbugfinder** [comment]
> @tbugfinder thanks for the PR. Could you please add valid integration tests for this ..

Unfortunately there isn't a terraform resource available. I checked `aws.tf` however I haven't seen a pattern how the project dealt in such cases (if there were even any).

**Review by sathish-progress — APPROVED** [review]
LGTM, we may need to have plural resource if needed at some point

**Comment by sathish-progress** [comment]
hey @tbugfinder, Could you please update the resource document link to [readme](https://github.com/inspec/inspec-aws/blob/master/README.md) file ?

**Comment by tbugfinder** [comment]
> hey @tbugfinder, Could you please update the resource document link to [readme](https://github.com/inspec/inspec-aws/blob/master/README.md) file ?

done

**Pull request merged**